### PR TITLE
gh-119981: use do { ... } while(0) in some multi-line macros

### PR DESCRIPTION
**Pull request by picnixz**
* Issue: gh-119981

### CODE REVIEW

File: Python/symtable.c [hunk 1]
```diff
@@ -70,6 +70,9 @@
 #define DUPLICATE_TYPE_PARAM \
 "duplicate type parameter '%U'"
 
+#define DUPLICATE_ARGUMENT \
+"duplicate argument '%U' in function definition"
+
```
Comment (nineteendo) [inline Python/symtable.c]: Please move this back:

```suggestion
```

Comment (picnixz) [reply Python/symtable.c]: Oh yes, it was something that I started working on but I forgot to cancel that move. Thanks!

File: Python/symtable.c [hunk 2]
```diff
@@ -610,7 +610,7 @@ error_at_directive(PySTEntryObject *ste, PyObject *name)
         return 0; \
     } \
     Py_DECREF(o); \
-}
+} while(0)
```
Comment (vstinner) [inline Python/symtable.c]: Can you please add one indentation level, so this line doesn't seem to be at the module level?

Comment (picnixz) [reply Python/symtable.c]: I've also put the `do {` on a separate line since it could be a bit clearer with the additional indentation.

**Comment by vstinner** [comment]
LGTM, I enabled auto-merge.

**Comment by picnixz** [comment]
I think you should add the skip news label for that (I don't really think I need a NEWS entry, or do I?)

**Pull request merged**